### PR TITLE
Fix config warnings and xfails

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

Silencing 4 warnings and fix 2 xfails for `test_config_*` tests. Warnings are to do with annotation intervals, e.g.:

```
_______________________ test_config_map_grid_cross_size ________________________
----------------------------- Captured stderr call -----------------------------
basemap [WARNING]: Your primary y annotation interval exceeds the secondary interval.
basemap [WARNING]: GMT expects it to be the other way around (primary annotations are closest to axis, secondary are further away)
basemap [WARNING]: Consider correcting your command - proceeding with your selections
```

The 2 xfails are to be fixed by updating to `@check_figures_equal()` instead of `@pytest.mark.mpl_image_compare`. Should we delete the corresponding baseline images https://github.com/GenericMappingTools/pygmt/blob/v0.2.1/pygmt/tests/baseline/test_config_font_annot.png and https://github.com/GenericMappingTools/pygmt/blob/v0.2.1/pygmt/tests/baseline/test_config_font_one.png too?

Fixes #

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/tests/test_config.py [hunk 1]
```diff
@@ -48,27 +46,40 @@ def test_config_font_one():
...
-    fig.basemap(T="mjBR+w5c+d-4.5+l")
-    return fig
+        fig_ref.basemap(R="0/9/0/9", J="C3/3/9c", Tm="jTL+w4c+d4.5+l")
+    fig_ref.basemap(Tm="jBR+w5c+d-4.5+l")
 
+    with config(FONT="8p,red"):
+        fig_test.basemap(
+            region=[0, 9, 0, 9], projection="C3/3/9c", compass="jTL+w4c+d4.5+l"
+        )
+    fig_test.basemap(compass="jBR+w5c+d-4.5+l")
```
Comment (weiji14) [inline pygmt/tests/test_config.py:59]: There is a weird thing going on here. The tests fail when using `fig_ref, fig_test = Figure(), Figure()`, but pass when structured like so:

```python
    fig_ref = Figure()
    with config(FONT="8p,red"):
        fig_ref.basemap(R="0/9/0/9", J="C3/3/9c", Tm="jTL+w4c+d4.5+l")
    fig_ref.basemap(Tm="jBR+w5c+d-4.5+l")

    fig_test = Figure()
    with config(FONT="8p,red"):
        fig_test.basemap(
            region=[0, 9, 0, 9], projection="C3/3/9c", compass="jTL+w4c+d4.5+l"
        )
    fig_test.basemap(compass="jBR+w5c+d-4.5+l")
```

Not sure if this has any relation to #733?

Comment (seisman) [reply pygmt/tests/test_config.py]: Yes, the same reason.

File: pygmt/tests/test_config.py [hunk 2]
```diff
@@ -36,10 +37,7 @@ def test_config():
     return fig
 
 
-@pytest.mark.xfail(
-    reason="Baseline image not updated to use earth relief grid in GMT 6.1.0",
-)
-@pytest.mark.mpl_image_compare
+@check_figures_equal()
 def test_config_font_one():
```
Comment (seisman) [inline pygmt/tests/test_config.py:41]: This test doesn't make sense to me.

We should check if `with config(FONT="8p,red"):` has the same effects as
```
with config(FONT_ANNOT_PRIMARY="8p,red", FONT_LABEL="8p,red", FONT_TITLE="8p,red", ...):
```

File: pygmt/tests/test_config.py [hunk 3]
```diff
@@ -48,27 +46,40 @@ def test_config_font_one():
...
+    fig_test.basemap(compass="jBR+w5c+d-4.5+l")
 
-@pytest.mark.xfail(
-    reason="Baseline image not updated to use earth relief grid in GMT 6.1.0",
-)
-@pytest.mark.mpl_image_compare
+    return fig_ref, fig_test
+
+
+@check_figures_equal()
```
Comment (seisman) [inline pygmt/tests/test_config.py:64]: This one also doesn't make sense.

File: pygmt/tests/test_config.py (hunk 2 above)
Comment (weiji14) [reply pygmt/tests/test_config.py:41]: Great suggestion, I've made the change in eeccc77220bfba2b354eb00f2f073c5b6952f9c1.

**Review by seisman — APPROVED** [review]
> Should we delete the corresponding baseline images https://github.com/GenericMappingTools/pygmt/blob/v0.2.1/pygmt/tests/baseline/test_config_font_annot.png and https://github.com/GenericMappingTools/pygmt/blob/v0.2.1/pygmt/tests/baseline/test_config_font_one.png too?

If I remember it correctly, there was a bug in compass (fixed in https://github.com/GenericMappingTools/gmt/pull/3329), so it means we will never be able to reproduce these two baselines images anymore. It's OK for me to delete them.

**Pull request merged**